content/blog/2021-04-19-experiment-refs.md

```diff
@@ -0,0 +1,276 @@
+---
+title: Git Custom References for ML Experiments
+date: 2021-04-19
+description: |
+  In DVC 2.0, we’ve introduced a new feature set aimed at simplifying the
+  versioning of lightweight ML experiments. In this post, we’ll dive into
+  how exactly these new experiments work.
+
+descriptionLong: |
+  In [DVC 2.0](/blog/dvc-2-0-release), we’ve introduced a new feature set
+  aimed at simplifying the versioning of lightweight ML experiments. In this
+  post, we’ll show how DVC leverages the power of Git references to track each
+  experiment, while also completely abstracting away the need for you to
+  manually manage a potentially unlimited number of Git feature branches or
+  tags.
+
+picture: 2021-04-19/experiment-refs.png
+pictureComment: Utilizing Custom Git References in DVC
+author: peter_rowlands
+commentsUrl: https://discuss.dvc.org/t/utilizing-custom-git-references-in-dvc/727
+tags:
+  - MLOps
+  - Git
+  - Experiments
+---
+
+One of the main features provided by DVC is the ability to version machine
+learning (ML) pipelines and experiments using Git commits. While this works very
+well for versioning mature projects and models, for projects under active
+development that may require generating hundreds of experiments or more in a
+single day, typical Git workflows can be difficult to work with. This type of
+rapid experimentation may appear to fit nicely with the concept of Git feature
+branches, but a Git repository with such large numbers of branches will
+eventually become too unwieldy to manage.
+
+In DVC 2.0, we’ve introduced a new feature set aimed at simplifying the
+versioning of lightweight ML experiments. DVC now provides a series of `dvc exp`
+commands which allow you to easily generate new experiments with modified
+hyperparameters, and to quickly compare their results. In this post, we’ll show
+how DVC leverages the power of Git references to track each experiment, while
+also completely abstracting away the need for you to manually manage a
+potentially unlimited number of Git feature branches or tags.
+
+_Note: This post mainly focuses on the “How?” side of DVC 2.0 experiments. For a
+great overview of the “What?” check out our
+[2.0 release post](/blog/dvc-2-0-release) and our
+[Get Started: Experiments](/doc/start/experiments) guide._
+
+## Experiments in DVC 2.0
+
+At the heart of the new experiments feature is the `dvc exp run` command.
+Whenever a pipeline is executed with `dvc exp run`, the results will be
+automatically tracked by DVC as a single “experiment”. DVC will track everything
+in your workspace as a part of the experiment, including unstaged changes made
+prior to execution. This means that DVC experiments can be used to test the
+result of changes to DVC-tracked data or pipeline parameters, as well as changes
+to Git-tracked code.
+
+![Example experiment run](/uploads/images/2021-04-19/exp-run.gif 'Example experiment run')
```

Comment:
The exp run takes a bit but the results (2nd `exp show` table) is only shown for like a second before the GIF restarts.